front/src/modules/pathfinding/components/Pathfinding/PathfindingV2.tsx

```diff
@@ -300,19 +300,35 @@ const Pathfinding = ({ pathProperties, setPathProperties }: PathfindingProps) =>
               await postPathProperties(pathPropertiesParams).unwrap();
 
             if (electrifications && geometry && operational_points) {
-              const pathStepsWihPosition = compact(pathSteps).map((step, i) => ({
-                ...step,
-                positionOnPath: pathfindingResult.path_items_positions[i],
-              }));
-              dispatch(updatePathSteps(pathStepsWihPosition));
-
               const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
                 operational_points,
                 geometry,
                 pathfindingResult.length
               );
 
-              const allVias = upsertViasInOPs(suggestedOperationalPoints, pathStepsWihPosition);
+              // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
+              const updatedPathSteps: PathStep[] = compact(pathSteps).map((step, i) => {
+                const correspondingOp = suggestedOperationalPoints.find(
+                  (suggestedOp) =>
+                    step &&
```

Comment:
Not sure this one is 100% necessary, can a null `step` ever happen?

Reply:
No you're right, as I now compact pathsteps first, the null steps are removed. I'll fix it.